### PR TITLE
add metrics_on_dashboards_folder w/ program and readme, update main r…

### PR DESCRIPTION
**Pull request by darthnater007**
…eadme

### CODE REVIEW

File: metrics_on_dashboards_report/metrics_on_dashboards_report.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#make sure the datadogpy library is downloaded in this machine or environment first
+from datadog import initialize,api
```
Comment (ericmustin) [inline metrics_on_dashboards_report/metrics_on_dashboards_report.py]: A comment is good here but you probably shouldn't assume the user is going to read the source code beside the readme, so you can bubble this up with a `try` block like below
```
#make sure the datadogpy library is downloaded in this machine or environment first
try:
    from datadog import initialize,api
except ImportError:
    print("please make sure the datadogpy library is downloaded in this machine or environment first, find install instructions at https://github.com/DataDog/datadogpy")
    quit()
```

File: metrics_on_dashboards_report/metrics_on_dashboards_report.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+options = {
+    'api_key' : api_key,
+    'app_key' : app_key
+}
+initialize(**options)
+print('Intialized!\n')
+
+#return every dashboard
+print('Getting All Screenboards\n')
+screen_resp = api.Screenboard.get_all()
```
Comment (ericmustin) [inline metrics_on_dashboards_report/metrics_on_dashboards_report.py]: invalid api or app key error responses get swallowed here and create unclear/implicit errors down the line, some error handling here on the response object would be nice, pretty straightforward response that you can do validation on `{u'errors': [u'Invalid API key or Application key']}`

File: metrics_on_dashboards_report/metrics_on_dashboards_report.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+print('All Screenboards Received!\n')
+print('Getting All Timeboards\n')
+time_resp = api.Timeboard.get_all()
```
Comment (ericmustin) [inline metrics_on_dashboards_report/metrics_on_dashboards_report.py]: ditto here, validation on `time_resp` would make this a lot easier to debug for a future user

File: metrics_on_dashboards_report/metrics_on_dashboards_report_readme.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+#Metrics on Dashboards Report
+
+This program is a simple console program written in Python 2.7.10, and utilizes the datadogpy API library to return a list of dashboards that have any metric in a series of user provided metrics present in their queries.
+
+This program requires that the datadogpy library be installed on the machine or environment on which the program is running.  Download and installation instructions can be found [here](https://github.com/DataDog/datadogpy)
+
+Once you have copied this program locally, it can be run in a terminal from the directory it is sotred in by running `python metrics_on_dashboards_report.py`.  Simply follow the on screen instructions, and receive a report of Dashboards where any of the metrics you provide are used.
```
Comment (ericmustin) [inline metrics_on_dashboards_report/metrics_on_dashboards_report_readme.md]: typo

Comment (ericmustin) [reply metrics_on_dashboards_report/metrics_on_dashboards_report_readme.md]: An example of this being run (w/ api/app key redacted) with appropriate input format would be helpful here.

File: metrics_on_dashboards_report/metrics_on_dashboards_report_readme.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+
+This program is for anyone wondering "where the heck am I using this metric on my Dashboards?"
+
+*Note*: The calls that this program makes to the API use deprecated libraries that Datadog still offers support for.  Efforts to rewrite this program with the new Dashboard API over the separate Timeboard and Screenboard API calls are not currently planned, though I may do this in the future.
```
Comment (ericmustin) [inline metrics_on_dashboards_report/metrics_on_dashboards_report_readme.md]: This work is great and adds a lot of value, this is a use case i've had before and I'm glad to see this PR! nice job here overall, left a few minor codestyle comment/suggestions.  I don't mind this getting merged but given that this API is getting deprecated it feels like a non zero amount of technical debt to add for people down the line, might be a good larger project in general to clean up this `Miscellany` repo to point to the new API

**Comment by ckelner** [comment]
Good stuff, thanks @darthnater007 !

**Pull request merged**